examples/demo_bgp.pp

```diff
@@ -130,46 +130,72 @@
   #---------------------------------------------------------------------------#
   # Configure BGP IPv4 Neighbors
   #---------------------------------------------------------------------------#
+
+  $capability_negotiation = $operatingsystem ? {
+    'nexus' => true,
+    default => undef
+  }
+  $dynamic_capability = $operatingsystem ? {
+    'nexus' => true,
+    default => undef
+  }
+  $log_neighbor_changes = $operatingsystem ? {
+    'nexus' => disable,
+    default => undef
+  }
+  $low_memory_exempt = $operatingsystem ? {
+    'nexus' => false,
+    default => undef
+  }
+  $remove_private_as = $operatingsystem ? {
+    'nexus' => 'all',
+    default => undef
+  }
+  $update_source = $operatingsystem ? {
+    'nexus' => 'ethernet1/1',
+    default => 'fastethernet1/1/1/1'
+  }
+
   cisco_bgp_neighbor {'55.77 blue 1.1.1.1':
     ensure                 => present,
 
     #Properties
     description            => 'my description',
     connected_check        => true,
-    capability_negotiation => true,
-    dynamic_capability     => true,
+    capability_negotiation => $capability_negotiation,
+    dynamic_capability     => $dynamic_capability,
     ebgp_multihop          => 2,
-    local_as               => 55.77,
-    log_neighbor_changes   => disable,
-    low_memory_exempt      => false,
+    local_as               => 55.88,
+    log_neighbor_changes   => $log_neighbor_changes,
+    low_memory_exempt      => $low_memory_exempt,
     remote_as              => 12,
-    remove_private_as      => 'all',
+    remove_private_as      => $remove_private_as,
     shutdown               => true,
     suppress_4_byte_as     => true,
     timers_keepalive       => 90,
     timers_holdtime        => 270,
-    update_source          => 'ethernet1/1',
+    update_source          => $update_source,
     transport_passive_only => false,
```

Comment:
I'm not a big fan of undocumented features - if it's working in XR, let's go ahead and document/test it. If it's not ready for that yet then we shouldn't be committing the code for that feature. IMHO.

Glenn

From: Github Notifications <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-puppet-module <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, December 8, 2015 at 10:28 AM
To: cisco/cisco-network-puppet-module <cisco-network-puppet-module@noreply.github.com<mailto:cisco-network-puppet-module@noreply.github.com>>
Cc: Glenn Matthews <glmatthe@cisco.com<mailto:glmatthe@cisco.com>>
Subject: Re: [cisco-network-puppet-module] BGP Neighbor puppet support for XR. (#101)


In examples/demo_bgp.pp<https://github.com/cisco/cisco-network-puppet-module/pull/101#discussion_r46966791>:

>      shutdown               => true,
>      suppress_4_byte_as     => true,
>      timers_keepalive       => 90,
>      timers_holdtime        => 270,
> -    update_source          => 'ethernet1/1',
> +    update_source          => $update_source,
>      transport_passive_only => false,


Since the main point of this story was functional equivalence to NX, I was going to just leave transport_passive_mode as an unadvertised feature until the story where we add XR-specific properties. I can be talked into fully documenting it and adding it to the beaker tests, though. (I did test it through beaker, I just removed it before I committed).

-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-puppet-module/pull/101/files#r46966791>.